lib/cli/run.js

```diff
@@ -0,0 +1,392 @@
+        group: GROUPS.OUTPUT,
+        requiresArg: true,
+        type: 'string'
+      },
+      reporters: {
+        description: 'List built-in reporters & exit',
+        type: 'boolean'
+      },
+      require: {
+        alias: 'r',
+        default: [],
+        defaultDescription: '(none)',
+        description: 'Require module',
+        group: GROUPS.FILES,
+        requiresArg: true,
+        type: 'array'
+      },
+      retries: {
+        description: 'Retry failed tests this many times',
+        group: GROUPS.RULES,
+        type: 'number'
+      },
+      slow: {
+        alias: 's',
+        default: 75,
+        description: 'Specify "slow" test threshold (in milliseconds)',
+        group: GROUPS.RULES,
+        type: 'number'
+      },
+      sort: {
+        alias: 'S',
+        description: 'Sort test files',
+        group: GROUPS.FILES,
+        type: 'boolean'
+      },
+      timeout: {
+        alias: 't',
+        default: 2000,
+        description: 'Specify test timeout threshold (in milliseconds)',
+        group: GROUPS.RULES
+      },
+      ui: {
+        alias: 'u',
+        default: 'bdd',
+        description: 'Specify user interface',
+        group: GROUPS.RULES,
+        requiresArg: true,
+        type: 'string'
+      },
+      watch: {
+        alias: 'w',
+        description: 'Watch files in the current working directory for changes',
+        group: GROUPS.FILES,
+        type: 'boolean'
+      }
+    })
+    .positional('spec', {
+      default: ['test/'],
+      description: 'One or more files, directories, or globs to test',
+      type: 'array'
+    })
+    .config('config', 'Load config file', filepath => {
+      // this is actually a yargs bug.
+      // remove once it's fixed
+      if (cachedConfig) {
+        debug('cached config hit');
+        return cachedConfig;
+      }
+      // yargs normalizes this into an absolute path, which is why
+      // this is weird.
+      if (filepath === path.resolve(CONFIG_PATH_PLACEHOLDER)) {
+        debug(`no config file specified`);
+        filepath = findConfig(cwd);
+      }
+      cachedConfig = loadConfig(filepath);
+      debug(`loaded config at ${filepath}`);
+      return cachedConfig;
+    })
+    .pkgConf('mocha')
+    .check(argv => {
+      // "one-and-dones"
+
+      if (argv[SHOW_HIDDEN_HELP_OPTION]) {
+        debug('hidden options:');
+        yargs.showHelp();
+        process.exit();
+      }
+
+      if (argv.interfaces) {
+        debug('interfaces:');
+        showInterfaces();
+        process.exit();
+      }
+
+      if (argv.reporters) {
+        debug('reporters:');
+        showReporters();
+        process.exit();
+      }
+
+      // yargs.implies() isn't flexible enough to handle this
+      if (argv.invert && !('fgrep' in argv || 'grep' in argv)) {
+        throw new Error(
+          '"--invert" requires one of "--fgrep <str>" or "--grep <regexp>"'
+        );
+      }
+
+      if (argv.compilers) {
```

Comment:
Do you need an unseen group `DEPRECATED` for retired options? How would `compilers` end up as an `argv` argument if it doesn't exist first?

Reply:
I think I meant `check()`. but this is done.